### ISSUE TITLE
Add provisioner label to OB,OBC,Secret,ConfigMap for automated cleanup

### ISSUE THREAD

**Issue by guymguym** [kube-object-storage/lib-bucket-provisioner#133]
The use case is when uninstalling a provisioner that is not running, we need a way to clean all of its resources.
Since all these resources OB,OBC,Secret,ConfigMap are not confined to any namespace, it would be best to add a label `provisioner=<provisioner-name>` to all of them.

**Comment by copejon** [comment]
+1 to unique labels.

This touches on a sticking point that's been brought up periodically since we settled on the library design.

We've kept coming back to a secondary operator running in the background - one per cluster.  @jeffvance and my own feeling on that was that we didn't want to ask library consumers to have to deploy and manage another operator in conjunction with their provisioner.   

Also, if the provisioner is uninstalled or crashes and resources are deleted by this supporting operator, it could quickly lead to bad state.  If the API resources are completely deleted while the provisioner is down, there's no way it can know to delete/suspend the backing bucket.

I think in this case, the answer is probably to implement the OBC finalizer mentioned in #131.
